src/pspm_sf_dcm.m

```diff
@@ -105,12 +105,13 @@
 y = scr;
 y = y - min(y);
 % 4.4 determine initial conditions
-x0 = y(1:3);
+y_non_nan = y(~isnan(y));
+x0 = y_non_nan(1:3);
 X0(1, 1)   = mean(x0);
 X0(2, 1)   = mean(diff(x0));
 X0(3, 1)   = diff(diff(x0));
 priors.muX0 = X0;
-nresp = floor(fresp * numel(y)/sr) + 1;
+nresp = floor(fresp * numel(y_non_nan)/sr) + 1;
```

Comment:
This line should be reverted. the numbe of modelled responses should not depend on the number of valid data points, only on the duration of the epoch.

Reply:
Understood. I updated this from `y` to `y_non_nan` because of the incorrect `x0` calculated at line 109 when the initial values are NaN. I do not have knowledge about whether `nresp` should use `y` or `y_non_nan`. I have reverted this line now.